configapi/api_default_test.go

```diff
@@ -266,6 +267,145 @@ func TestGetDeviceGroupByNameDoesExists(t *testing.T) {
 	}
 }
 
+func TestDeviceGroupDeleteHandler_DeviceGroupExistsInNetworkSlices(t *testing.T) {
+	gin.SetMode(gin.TestMode)
+	router := gin.Default()
+	AddConfigV1Service(router)
+
+	testCases := []struct {
+		name         string
+		route        string
+		dbAdapter    dbadapter.DBInterface
+		expectedCode int
+	}{
+		{
+			name:         "Device Group exists i",
+			route:        "/config/v1/device-group/group1",
+			dbAdapter:    &MockMongoClientManyNetworkSlices{},
+			expectedCode: http.StatusOK,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			dbadapter.CommonDBClient = tc.dbAdapter
+			origChannel := configChannel
+			configChannel = make(chan *configmodels.ConfigMessage, 10)
+			defer func() { configChannel = origChannel }()
+			req, err := http.NewRequest(http.MethodDelete, tc.route, nil)
+			if err != nil {
+				t.Fatalf("failed to create request: %v", err)
+			}
+			req.Header.Set("Content-Type", "application/json")
+			w := httptest.NewRecorder()
+
+			router.ServeHTTP(w, req)
+			if tc.expectedCode != w.Code {
+				t.Errorf("Expected `%v`, got `%v`", tc.expectedCode, w.Code)
+			}
+			expectedGroupName := "group1"
+			var msg *configmodels.ConfigMessage
+			expectedSliceNames := []string{"slice1", "slice2", "slice3"}
+			for sliceName := range expectedSliceNames {
```

Comment:
```suggestion
			for index := range expectedSliceNames {
```
This will only get the index of item in the slice of strings. 
 If we would like to get names, we can use 
index, name := range expectedSliceNames{..}

Reply:
you are absolutely right